include/cadmium/modeling/dynamic_model.hpp

```diff
@@ -0,0 +1,128 @@
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef CADMIUM_ATOMIC_HPP
+#define CADMIUM_ATOMIC_HPP
+
+#include <cadmium/modeling/dynamic_message_bag.hpp>
+
+namespace cadmium {
+    namespace dynamic {
+        namespace modeling {
+
+            struct EOC {
+                std::string _from;
+                std::shared_ptr<cadmium::dynamic::link_abstract> _link;
+
+                EOC() = delete;
+
+                EOC(std::string from, std::shared_ptr<cadmium::dynamic::link_abstract> l)
+                        : _from(from), _link(l) {}
+
+                EOC(const EOC& other)
+                        : _from(other._from), _link(other._link) {}
+            };
+
+            struct EIC {
+                std::string _to;
+                std::shared_ptr<cadmium::dynamic::link_abstract> _link;
+
+                EIC() = delete;
+
+                EIC(std::string to, std::shared_ptr<cadmium::dynamic::link_abstract> l)
+                        : _to(to), _link(l) {}
+
+                EIC(const EIC& other)
+                        : _to(other._to), _link(other._link) {}
+            };
+
+            struct IC {
+                std::string _to;
+                std::string _from;
+                std::shared_ptr<cadmium::dynamic::link_abstract> _link;
+
+                IC() = delete;
+
+                IC(std::string to, std::string from, std::shared_ptr<cadmium::dynamic::link_abstract> l)
+                        : _to(to), _from(from), _link(l) {}
+
+                IC(const IC& other)
+                        : _to(other._to), _from(other._from), _link(other._link) {}
+            };
+
+            using dynamic_EC=std::tuple<std::type_index, std::type_index, std::type_index>;
+            using dynamic_IC=std::tuple<std::type_index, std::type_index, std::type_index, std::type_index>;
+
+            using Ports = std::vector<std::type_index>;
+            using EICs = std::vector<EIC>;
+            using EOCs = std::vector<EOC>;
+            using ICs = std::vector<IC>;
+
+            using initilizer_list_Ports = std::initializer_list<std::type_index>;
+            using initializer_list_EOCs = std::initializer_list<EOC>;
+            using initializer_list_EICs = std::initializer_list<EIC>;
+            using initializer_list_ICs = std::initializer_list<IC>;
+
+            /**
+             * @brief Empty class to allow pointer based polymorphism between classes derived from atomic.
```

Comment:
Is this still being only used for atomic?

Reply:
No, this is used for both atomic and coupled models. 
I have extended the description to include the coupled models.